src/servers/src/postgres/types.rs

```diff
@@ -279,298 +279,345 @@ pub(super) fn parameters_to_scalar_values(
+                    ScalarValue::IntervalMonthDayNano(data.map(|i| Interval::from(i).to_i128()))
                 }
             }
             &Type::BYTEA => {
                 let data = portal.parameter::<Vec<u8>>(idx, &client_type)?;
-                match server_type {
-                    ConcreteDataType::String(_) => {
-                        ScalarValue::Utf8(data.map(|d| String::from_utf8_lossy(&d).to_string()))
-                    }
-                    ConcreteDataType::Binary(_) => ScalarValue::Binary(data),
-                    _ => {
-                        return Err(invalid_parameter_error(
-                            "invalid_parameter_type",
-                            Some(&format!(
-                                "Expected: {}, found: {}",
-                                server_type, client_type
-                            )),
-                        ));
+                if let Some(server_type) = &server_type {
+                    match server_type {
+                        ConcreteDataType::String(_) => {
+                            ScalarValue::Utf8(data.map(|d| String::from_utf8_lossy(&d).to_string()))
+                        }
+                        ConcreteDataType::Binary(_) => ScalarValue::Binary(data),
+                        _ => {
+                            return Err(invalid_parameter_error(
+                                "invalid_parameter_type",
+                                Some(&format!(
+                                    "Expected: {}, found: {}",
+                                    server_type, client_type
+                                )),
+                            ));
+                        }
                     }
+                } else {
+                    ScalarValue::Binary(data)
                 }
             }
             &Type::JSONB => {
                 let data = portal.parameter::<serde_json::Value>(idx, &client_type)?;
-                match server_type {
-                    ConcreteDataType::Binary(_) => {
-                        ScalarValue::Binary(data.map(|d| jsonb::Value::from(d).to_vec()))
-                    }
-                    _ => {
-                        return Err(invalid_parameter_error(
-                            "invalid_parameter_type",
-                            Some(&format!(
-                                "Expected: {}, found: {}",
-                                server_type, client_type
-                            )),
-                        ));
+                if let Some(server_type) = &server_type {
+                    match server_type {
+                        ConcreteDataType::Binary(_) => {
+                            ScalarValue::Binary(data.map(|d| jsonb::Value::from(d).to_vec()))
+                        }
+                        _ => {
+                            return Err(invalid_parameter_error(
+                                "invalid_parameter_type",
+                                Some(&format!(
```

Comment:
nitpick: we can change the `detail` argument in `invalid_parameter_error` from `Option<&str>` to `Option<String>` to avoid a clone.